src/components/table/table.js

```diff
@@ -56,12 +56,15 @@ class Table extends PureComponent {
 
   componentWillReceiveProps(nextProps) {
     const { data, defaultColumns } = this.props;
-    if (nextProps.data !== data) {
-      this.setState({ data: nextProps.data });
+    const { data: nextData, defaultColumns: nextDefaultColumns } = nextProps;
+    if (nextData !== data) {
+      this.setState({ data: nextData });
     }
-    if (nextProps.defaultColumns !== defaultColumns) {
+    if (nextDefaultColumns !== defaultColumns) {
```

Comment:
I'm not sure if this bit works as expected. I assume we would like to avoid updating `deafultColumns` prop when these two arrays are identical. If so, I think you need to check the content of the array. 

Let's take an example:
![image](https://user-images.githubusercontent.com/15097138/70528281-c0d37b80-1b45-11ea-9797-9549a490185b.png)
I'd suggest using [isEqual](https://lodash.com/docs/4.17.15#isEqual) function from lodash which provides a deep comparison between two arrays/objects/values. Also please check line 60 `if (nextData !== data) {`, it can be a similar case :wink:

Reply:
You are right. I just pushed the change